### PR TITLE
Support the "--library-makefile" flag when compiling with LLVM

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
This is a simple fix which moves the branch to generate a Makefile as per `--library-makefile` so that it is common code shared by both backends.

So far all tests in `test/interop/C/llvm/makefiles` have passed, and so I've removed the futures for them.

I've placed the new code in a branch that only fires when `llvmCodegen` is `true`. I've also removed the `bad` and `future` files in `test/interop/C/llvm/makefiles`, and one in `test/interop/C/llvm/noLibFlag`. 

If tests with `CHPL_LLVM=none` and `CHPL_LLVM=llvm` both pass, then I will consider removing them from the `llvm` subdirectory and remove the `CHPL_LLVM=none` from the SKIPIF.

**Testing:**

- [x] All tests on `linux64` when `CHPL_LLVM=none`
- [x] All tests on `linux64` when `CHPL_LLVM=llvm`

### CODE REVIEW

**Comment by lydia-duncan** [comment]
This seems reasonable, though I'm impressed that changes to the generated Makefile weren't needed (my thinking was that it would be a blanket addition for LLVM only rather than something that depended on the C compiler being used for linking, since it was our LLVM set up that made it necessary and it doesn't seem to cause harm to have it listed twice).

What testing did you run?

**Comment by dlongnecke-cray** [comment]
Thanks for the helpful discussion offline!

I've gone ahead and added a workaround that covers the GNU linker. All tests for the `--library-makefile` flag in `test/interop/C/llvm/makefiles` now pass on `linux64`, so I went ahead and removed the futures.

Let me know if there's anything else you'd like me to change.

File: compiler/codegen/library.cpp [hunk 1]
```diff
@@ -226,7 +227,16 @@ static void printMakefileLibraries(fileinfo makefile, std::string name) {
...
+
+  // Erase trailing newline if needed for the workaround below.
+  if (libraries.size() > 0 && *libraries.rbegin() == '\n') {
+    libraries.erase(libraries.end() - 1);
+  }
+
+  fprintf(makefile.fptr, " %s", libraries.c_str());
+
+  // GNU linker won't be able to see config symbols without this.
+  fprintf(makefile.fptr, " %s\n\n", libname.c_str());
```
Comment (lydia-duncan) [inline compiler/codegen/library.cpp]: I would like this to only occur for LLVM, if possible.

Also, since you are now modifying something that gets called by both backends, you need to test both backends - which means you need to run test/interop/C/makefiles with CHPL_LLVM=none

File: test/interop/C/llvm/makefiles/commandLineLib.future [hunk 2]
```diff
@@ -1 +0,0 @@
-bug: support --library-makefile with LLVM
```
Comment (lydia-duncan) [inline test/interop/C/llvm/makefiles/commandLineLib.future]: This test has a .bad file that should also get deleted with the .future file

File: test/interop/C/llvm/makefiles/requireStmt.future [hunk 3]
```diff
@@ -1 +0,0 @@
-bug: support --library-makefile with LLVM
```
Comment (lydia-duncan) [inline test/interop/C/llvm/makefiles/requireStmt.future]: I believe there is one more future that is missing.  It lives in the test/interop/C/llvm/noLibFlag directory (which checks the behavior of the `--library-*` flags when `--library` itself is omitted.  Basically, the intention is that these flags will turn on `--library` compilation if it is not explicitly turned on, since the flag names are long and something a user would only want with `--library` compilation).

Note that when checking the C backend as mentioned in my earlier comment, you will also want to run the C equivalent (so test/interop/C/noLibFlag)

**Comment by lydia-duncan** [comment]
Also, please update your PR description with the testing you have run

File: test/interop/C/llvm/makefiles/commandLineLib.future (hunk 2 above)
Comment (dlongnecke-cray) [reply test/interop/C/llvm/makefiles/commandLineLib.future]: Great thanks.

File: compiler/codegen/library.cpp (hunk 1 above)
Comment (dlongnecke-cray) [reply compiler/codegen/library.cpp]: Are the tests in the `../makefiles` directory sufficient for this? I was under the impression they are equivalent except one runs with the `--llvm` flag.

Comment (lydia-duncan) [reply compiler/codegen/library.cpp]: Yes and no.  The directory itself has a skipif for CHPL_LLVM != none, so they won't normally run in that setting (even though they should be equivalent).  And I'd rather be a little cautious than rely on an assumption that is false

File: compiler/codegen/library.cpp [hunk 4]
```diff
@@ -228,17 +228,22 @@ static void printMakefileLibraries(fileinfo makefile, std::string name) {
...
+    libraries.erase(libraries.end() -1);
   }
 
-  fprintf(makefile.fptr, " %s", libraries.c_str());
-
-  // GNU linker won't be able to see config symbols without this.
-  fprintf(makefile.fptr, " %s\n\n", libname.c_str());
+  // Append the Chapel library as the last linker argument.
+  fprintf(makefile.fptr, " %s %s\n\n", libraries.c_str(), libname.c_str());
 }
```
Comment (lydia-duncan) [inline compiler/codegen/library.cpp]: I'd format this as an if/else rather than "if { return }; if (other condition)".  Someone might otherwise accidentally remove the return statement later without necessarily realizing the code that followed shouldn't run in the other situation

File: compiler/codegen/library.cpp [hunk 5]
```diff
@@ -228,17 +228,22 @@ static void printMakefileLibraries(fileinfo makefile, std::string name) {
...
 
+
+
```
Comment (lydia-duncan) [inline compiler/codegen/library.cpp]: Extra lines?

File: compiler/codegen/library.cpp (hunk 4 above)
Comment (dlongnecke-cray) [reply compiler/codegen/library.cpp]: Ok.

**Comment by dlongnecke-cray** [comment]
Sorry for the delay on the testing update, the compiler is having a rather painfully slow time building right now.

**Comment by dlongnecke-cray** [comment]
All tests except `gsl_demo` have passed on `linux64` when `CHPL_LLVM=llvm`.

**Comment by lydia-duncan** [comment]
👍
